Constrain dependency versions

reiterative opened the pull request:
Constrain the project's dependency versions so that newer versions do not introduce breaking changes. This is required because of a recent update to plantuml-markdown, which breaks Doorstop.

Also removes redundant pre-commit config file

lbiaggi commented on pyproject.toml:
```diff
@@ -45,15 +45,15 @@ classifiers = [
 
 python = "^3.10"
```
Can you restrict this to be 3.10 <= python <= 3.13? Or even  < 3.13.

reiterative replied:
Done.

reiterative committed:
Constrain dependency versions

Constrain the project's dependency versions so that newer versions
do not introduce breaking changes. This is required because of a
recent update to plantuml-markdown, which breaks Doorstop.

Also removes redundant pre-commit config file
